### ISSUE TITLE
Seriously how do we use this? A simple example please

### ISSUE THREAD

**Issue by FurkanGozukara** [Mr1Penguin/DeviantartApi#12]
I have downloaded the library from nuget package into .net 4.8 WPF application

I have got my client id and client secret from here : https://www.deviantart.com/developers/apps

Ok then how do i authentica and then search a keyword

I want to search keyword fakemon and fetch all results with pagination

Thank you

**Comment by Mike-E-angelo** [comment]
Did you figure this out @FurkanGozukara?  I am struggling to see how to simply download a whois.  Seems like a SignIn is required, but it is not part of the ExecuteAsync call...

**Comment by Mike-E-angelo** [comment]
There also appears to be a Callback Uri but I am not seeing how to use that.  I basically have my client/secret and want to make a whois call.

**Comment by FurkanGozukara** [comment]
> Did you figure this out @FurkanGozukara? I am struggling to see how to simply download a whois. Seems like a SignIn is required, but it is not part of the ExecuteAsync call...

I am using old school software right now

I am doing http posts and  JsonConvert.DeserializeObject response

**Comment by Mike-E-angelo** [comment]
Yep @FurkanGozukara I unfortunately had to resort to the same, via `IHttpClientFactory`.  Here's my commit in case anyone else is looking for something a bit more current for some reference:
https://github.com/DragonSpark/Framework/commit/ca6a407f2515fc7b4f2c59a2aa94a7b45cb80970

In my case, I am querying the user object based on user name to get their GUID/identity.

What's interesting is that the `whois` endpoint is protected and cannot be accessed by `client_credentials` grant, whereas the `user/profile` can.  This really doesn't make sense to me as the `user/profile` contains wayyyyyy more information than `whois`. 🤷‍♂️  But anyways, I am glad this is over. :P  Onto the next fire.
